### PR TITLE
Support parallel for `all?` command

### PR DESCRIPTION
**Pull request by Kangaxx-0**
# Description

Parallelism support for internal command `all` as requested from #6061 

TODO - I will add parallelism support for other commands in different PRs

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

**Comment by fdncred** [comment]
oh, nice! you come up with some interesting PRs!

File: crates/nu-command/src/filters/all.rs [hunk 1]
```diff
@@ -66,31 +78,79 @@ impl Command for All {
...
+                        &engine_state,
+                        &mut stack,
+                        block,
+                        PipelineData::new(span),
+                        call.redirect_stdout,
+                        call.redirect_stderr,
+                    ) {
+                        Err(error) => Value::Error { error },
+                        Ok(pipeline_data) => {
+                            if !pipeline_data.into_value(span).is_true() {
```
Comment (WindSoilder) [inline crates/nu-command/src/filters/all.rs:101]: to me it seems something like: `if !false {return true} else {return false}`...It's not really readable

Maybe we can just do something like this?
```rust
Ok(pipeline_data) => {
    Value::Bool { val: pipeline_data.into_value(span).is_true(), span }
}
```

Comment (Kangaxx-0) [reply crates/nu-command/src/filters/all.rs]: Thanks for the catching!

**Comment by jntrnr** [comment]
I'm doing some refactor of the engine which might make this a bit harder. We may want to hold off on adding more parallel commands until we get a good idea of how it will change how parallel code works in the engine.

**Comment by fdncred** [comment]
making this a draft considering jt's comments.

**Comment by fdncred** [comment]
closing this in favor of waiting to see if it's experiments pan out.

**Pull request closed without merging**